CNTR0020E caused by java.lang.NoClassDefFoundError: com/ibm/ejs/container/util/ExceptionUtil

**Describe the bug**  
For an Enterprise Bean (EJB) configured as a JAX-WS Endpoint, the following error will be reported if an EJB interceptor throws an exception that is not on the throws clause of the EJB method, not a RuntimeException, and not an Error.

```
CNTR0020E: EJB threw an unexpected (non-declared) exception during invocation of method "xxxxxx" on bean "BeanId(appA#moduleB.war#beanC, null)". Exception data: java.lang.NoClassDefFoundError: com/ibm/ejs/container/util/ExceptionUtil
	at xx.yy.WSEJBProxySLbeanC_54321c678.xxxxxx(WSEJBProxySLbeanC_54321c678.java)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.apache.cxf.service.invoker.AbstractInvoker.performInvocation(AbstractInvoker.java:184)
	at com.ibm.ws.jaxws.ejb.EJBMethodInvoker.performInvocation(EJBMethodInvoker.java:107)
Caused by: java.lang.ClassNotFoundException: com.ibm.ejs.container.util.ExceptionUtil
	at com.ibm.ws.classloading.internal.AppClassLoader.findClassCommonLibraryClassLoaders(AppClassLoader.java:751)
	at com.ibm.ws.classloading.internal.AppClassLoader.findClass(AppClassLoader.java:327)
	at com.ibm.ws.classloading.internal.AppClassLoader.findOrDelegateLoadClass(AppClassLoader.java:705)
	at com.ibm.ws.classloading.internal.AppClassLoader.loadClass(AppClassLoader.java:584)
	at com.ibm.ws.classloading.internal.AppClassLoader.loadClass(AppClassLoader.java:553)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
```

**Steps to Reproduce**  
Install a JAX-WS application that uses an EJB as the endpoint, and include an interceptor that will throw an exception that is not on the throws clause of the EJB interface.  Then invoke that JAX-WS endpoint.

**Expected behavior**  
Invoking the JAX-WS Endpoint should result in an EJBException that has the exception thrown by the interceptor as the cause.

**Diagnostic information:**  
 - OpenLiberty Version: all
 - Affected feature(s): jaxws-2.2, xmlWS-3.0, xmlWS-4.0, ejbLite-3.2, enterpriseBeans-4.0
 - Java Version: any

**Additional context**  
None

--- a/dev/com.ibm.ws.ejbcontainer.core/src/com/ibm/ejs/container/WSEJBWrapper.java
+++ b/dev/com.ibm.ws.ejbcontainer.core/src/com/ibm/ejs/container/WSEJBWrapper.java
@@ -1,10 +1,10 @@
 /*******************************************************************************
- * Copyright (c) 2008, 2012 IBM Corporation and others.
+ * Copyright (c) 2008, 2024 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License 2.0
  * which accompanies this distribution, and is available at
  * http://www.eclipse.org/legal/epl-2.0/
- * 
+ *
  * SPDX-License-Identifier: EPL-2.0
  *
  * Contributors:
@@ -15,6 +15,8 @@ package com.ibm.ejs.container;
 import java.lang.reflect.Method;
 import java.rmi.RemoteException;
 import java.util.Map;
+
+import javax.ejb.EJBException;
 
 import com.ibm.ejs.container.util.ExceptionUtil;
 import com.ibm.websphere.ras.Tr;
@@ -478,4 +480,21 @@ public class WSEJBWrapper extends EJSWrapperBase implements WSEJBEndpointManager
         return hasAroundInvoke;
     }
 
+    /**
+     * Returns an EJBException with the message text "See nested exception"
+     * and the specified cause, or the cause itself, if it is already an
+     * EJBException. <p>
+     *
+     * Included in this IBM API "internal" interface to provide the generated
+     * WSEJBProxy implementations access to EJB container internal code that
+     * provides this exception handling behavior. <p>
+     *
+     * @param cause the cause of the EJBException. If already an EJBException
+     *            it will just be returned; null is permitted.
+     * @returns an EJBException with appropriate nested exception, stack,
+     *          and message text.
+     **/
+    public static EJBException EJBException(Throwable cause) {
+        return ExceptionUtil.EJBException(cause);
+    }
 }

--- a/dev/com.ibm.ws.ejbcontainer.core/src/com/ibm/ws/ejbcontainer/jitdeploy/WSEJBProxy.java
+++ b/dev/com.ibm.ws.ejbcontainer.core/src/com/ibm/ws/ejbcontainer/jitdeploy/WSEJBProxy.java
@@ -1,10 +1,10 @@
 /*******************************************************************************
- * Copyright (c) 2008, 2009 IBM Corporation and others.
+ * Copyright (c) 2008, 2024 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License 2.0
  * which accompanies this distribution, and is available at
  * http://www.eclipse.org/legal/epl-2.0/
- * 
+ *
  * SPDX-License-Identifier: EPL-2.0
  *
  * Contributors:
@@ -33,10 +33,26 @@ import static com.ibm.ws.ejbcontainer.jitdeploy.JITUtils.getTypes;
 import static com.ibm.ws.ejbcontainer.jitdeploy.JITUtils.unbox;
 import static com.ibm.ws.ejbcontainer.jitdeploy.JITUtils.writeToClassFile;
 
-import static org.objectweb.asm.Opcodes.*;
 import static com.ibm.ws.ejbcontainer.jitdeploy.JITUtils.INDENT;
 import static com.ibm.ws.ejbcontainer.jitdeploy.JITUtils.TYPE_Exception;
 import static com.ibm.ws.ejbcontainer.jitdeploy.JITUtils.TYPE_Object_ARRAY;
+
+import static org.objectweb.asm.Opcodes.AASTORE;
+import static org.objectweb.asm.Opcodes.ACC_PUBLIC;
+import static org.objectweb.asm.Opcodes.ACC_SUPER;
+import static org.objectweb.asm.Opcodes.ACONST_NULL;
+import static org.objectweb.asm.Opcodes.ALOAD;
+import static org.objectweb.asm.Opcodes.ANEWARRAY;
+import static org.objectweb.asm.Opcodes.ATHROW;
+import static org.objectweb.asm.Opcodes.DUP;
+import static org.objectweb.asm.Opcodes.GETFIELD;
+import static org.objectweb.asm.Opcodes.GOTO;
+import static org.objectweb.asm.Opcodes.INVOKESPECIAL;
+import static org.objectweb.asm.Opcodes.INVOKESTATIC;
+import static org.objectweb.asm.Opcodes.INVOKEVIRTUAL;
+import static org.objectweb.asm.Opcodes.NEW;
+import static org.objectweb.asm.Opcodes.RETURN;
+import static org.objectweb.asm.Opcodes.V1_2;
 
 /**
  * Provides Just In Time runtime deployment of WebService Endpoint EJB Proxys. <p>
@@ -535,19 +551,19 @@ public final class WSEJBProxy
         // -----------------------------------------------------------------------
         // Bean instance could not possibly have thrown this (unless Throwable
         // itself), so this must be coming from a poorly behaving interceptor.
-        // This is a spec violation and cannot be reported dirctly.  Do the best
+        // This is a spec violation and cannot be reported directly.  Do the best
         // we can, and wrap it in an EJBException (i.e. RuntimeException).
         //
         // catch (Throwable <ex>)
         // {
-        //   throw ExceptionUtil.EJBException( <ex> );
+        //   throw WSEJBWrapper.EJBException( <ex> );
         // }
         // -----------------------------------------------------------------------
         Label main_catch_throwable = new Label();
         mg.visitLabel(main_catch_throwable);
         mg.storeLocal(caught_ex);
         mg.loadLocal(caught_ex);
-        mg.visitMethodInsn(INVOKESTATIC, "com/ibm/ejs/container/util/ExceptionUtil",
+        mg.visitMethodInsn(INVOKESTATIC, "com/ibm/ejs/container/WSEJBWrapper",
                            "EJBException", "(Ljava/lang/Throwable;)Ljavax/ejb/EJBException;");
         mg.visitInsn(ATHROW);
 

